prime-router/src/test/kotlin/history/azure/SubmissionFunctionTests.kt

```diff
@@ -514,60 +521,103 @@ class SubmissionFunctionTests : Logging {
         assertThat(response.status).isEqualTo(HttpStatus.NOT_FOUND)
     }
 
+    private fun setupEtorTestData(): UUID {
+        var receiveReportId: UUID = UUID.randomUUID()
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val action = Action()
+            action.actionName = TaskAction.receive
+            action.sendingOrg = "test org"
+            val receiveActionId = ReportStreamTestDatabaseContainer.testDatabaseAccess.insertAction(txn, action)
+            val receiveReport = ReportFile().setSchemaTopic(Topic.ETOR_TI).setReportId(UUID.randomUUID())
+                .setActionId(receiveActionId).setSchemaName("schema").setBodyFormat("hl7").setItemCount(1)
+            ReportStreamTestDatabaseContainer.testDatabaseAccess.insertReportFile(
+                receiveReport, txn, action
+            )
+
+            receiveReportId = receiveReport.reportId
+
+            action.actionName = TaskAction.convert
+
+            val convertActionId = ReportStreamTestDatabaseContainer.testDatabaseAccess.insertAction(txn, action)
+            val convertReport = ReportFile().setSchemaTopic(Topic.ETOR_TI).setReportId(UUID.randomUUID())
```

Comment:
Nit: could consider extracting this as a helper just to make the setup a little clearer